### PR TITLE
feat(cni): genServicePort

### PR DESCRIPTION
**Pull request by qingwusunny**
genServicePort resource according to endpoints resource

### CODE REVIEW

File: pkg/controller/k8s/endpoints_controller.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+		if err := r.Client.Update(ctx, update[i]); err != nil {
+			klog.Errorf("Failed to update ServicePort %+v, err: %s", *update[i], err)
+			return err
+		}
+	}
+
+	for i := range delete {
+		if err := r.Client.Delete(ctx, delete[i]); err != nil {
+			klog.Errorf("Failed to delete ServicePort %+v, err: %s", *delete[i], err)
+		}
```
Comment (zwtop) [inline pkg/controller/k8s/endpoints_controller.go:144]: `return err`

File: pkg/apis/service/v1alpha1/types.go [hunk 2]
```diff
@@ -21,11 +21,16 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	LabelRefEndpoints = "endpoints.service.everoute.io"
```
Comment (zwtop) [inline pkg/apis/service/v1alpha1/types.go:25]: What's the difference between `LabelRefEndpoints` and `.spec.svcRef`?

File: cmd/everoute-controller/main.go [hunk 3]
```diff
@@ -74,6 +75,7 @@ func main() {
 	flag.StringVar(&leaderElectionNamespace, "leader-election-namespace", "", "The namespace in which the leader election configmap will be created.")
 	flag.IntVar(&serverPort, "port", 9443, "The port for the Everoute controller to serve on.")
 	flag.BoolVar(&enableCNI, "enable-cni", false, "Enable CNI related controller.")
+	flag.BoolVar(&enableProxy, "enable-proxy", false, "Enable CNI service proxy")
```
[the remark this answers is not in the document]
Comment (qingwusunny) [reply cmd/everoute-controller/main.go:78]: yes

File: pkg/apis/service/v1alpha1/types.go (hunk 2 above)
Comment (qingwusunny) [reply pkg/apis/service/v1alpha1/types.go:25]: the value is same, but these meaning is different, it's better to save all for me

File: pkg/controller/k8s/endpoints_controller.go (hunk 1 above)
Comment (qingwusunny) [reply pkg/controller/k8s/endpoints_controller.go]: done

**Review by wu-cl — APPROVED** [review]

**Review by KeyongSun — APPROVED** [review]

**Pull request merged**